### PR TITLE
feat(cve): cli cve diff

### PR DESCRIPTION
**Pull request by laurentiuNiculae**
**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/extensions/search/cve/cve.go [hunk 1]
```diff
@@ -329,7 +331,21 @@ func getConfigAndDigest(metaDB mTypes.MetaDB, manifestDigestStr string) (ispec.I
...
+			continue
+		}
+
+		if cve.ContainsStr(searchedCVE) {
+			pageFinder.Add(cve)
+		}
+	}
+}
+
+func filterCVEList(cveMap []cvemodel.CVE, searchedCVE, excludedCVE string, pageFinder *CvePageFinder) {
```
Comment (andaaron) [inline pkg/extensions/search/cve/cve.go:348]:
```suggestion
func filterCVEList(cveList []cvemodel.CVE, searchedCVE, excludedCVE string, pageFinder *CvePageFinder) {
```

File: pkg/extensions/search/resolver.go [hunk 2]
```diff
@@ -277,6 +277,264 @@ func getCVEListForImage(
...
+		return false
+	}
+
+	arch := getArch(platform)
+
+	if arch == deref(platformInput.Arch, "") {
+		return true
+	}
+
+	return true
```
Comment (andaaron) [inline pkg/extensions/search/resolver.go:526]: Please double check if this should be true or false.

**Comment by laurentiuNiculae** [comment]
Example:
zli-linux-amd64 cve diff --url http://127.0.0.1:5000/ -u laur:password python:3.8.16-alpine3.17 python:3.8.16-alpine3.16

Should print CVE's in python:3.8.16-alpine3.17 and not in python:3.8.16-alpine3.16

File: pkg/extensions/search/resolver.go [hunk 3]
```diff
@@ -279,6 +279,260 @@ func getCVEListForImage(
...
+	}
+
+	return &gql_generated.Platform{
+		Os:   platform.Os,
+		Arch: platform.Arch,
+	}
+}
+
+// rename idea: identify image from input.
+func checkImageInput(ctx context.Context, imageInput gql_generated.ImageInput, metaDB mTypes.MetaDB,
```
Comment (andaaron) [inline pkg/extensions/search/resolver.go:449]: Definitely needs a rename.
 
```suggestion
func resolveImageData(ctx context.Context, imageInput gql_generated.ImageInput, metaDB mTypes.MetaDB,
```

Or

```suggestion
func resolveImageDigest(ctx context.Context, imageInput gql_generated.ImageInput, metaDB mTypes.MetaDB,
```

File: pkg/extensions/search/resolver.go [hunk 4]
```diff
@@ -279,6 +279,260 @@ func getCVEListForImage(
...
+		return nil
+	}
+
+	return &gql_generated.Platform{
+		Os:   platform.Os,
+		Arch: platform.Arch,
+	}
+}
+
+// rename idea: identify image from input.
```
Comment (andaaron) [inline pkg/extensions/search/resolver.go:448]: Remove after rename.

**Review by andaaron — COMMENTED** [review]
Except the rename it looks good. Please add more tests to increase coverage

**Review by andaaron — APPROVED** [review]

**Pull request merged**